documentation/general/design/dash-holistic-view.md

```diff
@@ -0,0 +1,627 @@
+plane, when a customer creates an operation from the cloud portal (for example a
+VNET creation), the controller allocates the resources, placement management,
+capacity, etc. via the  **NorthBound interface APIs**.
+
+##### SDN and DPU High-Availability (HA)
+
+High Availability (HA) is important for hardware failover incidents. The SDN
+controller selects a pair of cards and configures them identically.  
+The SDN controller will set up the L3 routes through the ToR switch levels,
+possibly even T2 levels for a third backup appliance or smart ToR and configure
+the HA "Peers." 
+
+> [!NOTE] HA sync protocols are WIP and will be standardized and interoperable,
+> not vendor-specified.
+
+The only requirement on the card from the HA perspective is for the cards to
+setup a channel between themselves for flow synchronization.  The
+synchronization mechanism is left for technology suppliers to define and
+implement. For more information, see [High Availability and
+Scale](../../high-avail/design/high-availability-and-scale.md) document.
+
+
+### DASH container
+
+The SDN controller communicates with a DASH device through a
+**[gNMI](https://github.com/Azure/DASH/wiki/Glossary#gnmi) endpoint** served by
+a new DASH SDN agent **running inside a new SONiC DASH container**.  
+
+In summary:
+
+- The DASH container translates SDN configuration modeled in gNMI into
+  **DASH_APP_DB** objects. The gNMI schema is closely related to the DASH DB
+  schema so in effect, the gNMI server is a thin RPC shim layer to the DB.
+- The **SONiC orchagent** inside the Switch State Service (SWSS) Container will
+  be enhanced to transform and translate these objects into **ASIC_DB objects**,
+  including the new **DASH-specific SAI objects**.  
+- An **enhanced syncd** will then configure the dataplane using the **technology
+  supplier-specific SAI library**.
+
+A **gNMI schema** will manage the following DASH components:
+
+- Elastic Network Interface (ENI)
+- Access Control Lists (ACLs)
+- Routing and mappings
+- Encapsulations
+- Other  
+
+> [!NOTE] See also work in progress PR [doc-ref-config-example
+> #93](https://github.com/Azure/DASH/pull/93).
+
+#### Multiple DPUs device
+
+In the case of a multiple DPUs device the following applies:
+
+- Each DPU provides a gNMI endpoint for SDN controller through a unique IP
+  address.
+- An appliance or smart switch containing multiple DPUs will contain multiple
+  gNMI endpoints for SDN controller, and the controller treats each DPU as a
+  separate entity.
+- To conserve IPv4 addresses, such an appliance or switch *might* contain a
+  proxy (NAT) function to map a single IP address:port combination into multiple
+  DPU endpoints with unique IPv4 addresses.  
+- No complex logic will run on the switches (switches do not have a top-level
+  view of other/neighboring switches in the infrastructure).
+
+> [!NOTE] Check w/SDN and SONiC if the controller communicates with the DPU, 
```

Comment:
In this model, AFAIK, each DPU shall have an instance of Sonic NOS that exposes the gNMI APIs to the SDN controller

Reply:
Fixed.